modules/asset/group/tests/src/Kernel/GroupTest.php

```diff
@@ -174,6 +174,16 @@ public function testGroupMembership() {
     // Assert that the animal's cache tags were not invalidated.
     $this->assertEntityTestCache($animal, TRUE);
 
+    // Change the log status to "abandoned" and confirm that the group
+    // membership is unchanged.
+    $second_log->set('status', 'abandoned');
+    $second_log->save();
+    $this->assertEquals($first_group->id(), $this->groupMembership->getGroup($animal)[0]->id(), 'Abandoned group assignment logs do not affect membership.');
+    $this->assertCorrectAssets([], $this->groupMembership->getGroupMembers([$second_group]), TRUE, 'Groups with only abandoned membership logs have zero members.');
+
+    // Assert that the animal's cache tags were not invalidated.
+    $this->assertEntityTestCache($animal, TRUE);
+
```

Comment:
This test covers changing a group membership log from `pending` -> `abandoned`.

It might be worth adding another test to cover creating a new `abandoned` log (expect no cache invalidation) or even changing from `done` -> `abandoned` (expect cache invalidation)

Reply:
Good idea. I'll add these and amend the relevant commit.